File: features/main/src/main/java/com/inquallity/daggersandbox/main/di/MainComponent.java
package com.inquallity.daggersandbox.main.di;

import com.inquallity.daggersandbox.core.crossmodule.CrossModule;
import com.inquallity.daggersandbox.core.domain.MainInteractor;
import com.inquallity.daggersandbox.main.MainActivity;

import dagger.BindsInstance;
import dagger.Component;

@Component
public interface MainComponent {

    void inject(MainActivity target);

    @Component.Builder
    public interface Builder {

        @BindsInstance
        Builder interactor(MainInteractor i);

        @BindsInstance
        Builder crossmodule(CrossModule c);

        MainComponent build();
    }
}

File: features/main/src/main/java/com/inquallity/daggersandbox/main/di/MainComponent.kt
package com.inquallity.daggersandbox.main.di

import com.inquallity.daggersandbox.core.crossmodule.CrossModule
import com.inquallity.daggersandbox.core.domain.MainInteractor
import com.inquallity.daggersandbox.main.MainActivity
import dagger.BindsInstance
import dagger.Component

@Component
interface MainComponent {

    fun inject(m: MainActivity)

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun interactor(i: MainInteractor): Builder

        @BindsInstance
        fun crossmodule(c: CrossModule): Builder
        fun build(): MainComponent
    }
}